### ISSUE TITLE
Linkify strips href attribute from exsiting relative links

### ISSUE THREAD

**Issue by alexington** [mozilla/bleach#153]
The `bleach.linkify()` method removes the href attribute of existing links with relative paths.

Absolute path link:

```
>>> bleach.linkify('<a href="http://google.com">example</a>')
u'<a href="http://google.com" rel="nofollow">example</a>'
```

Relative path link:

```
>>> bleach.linkify('<a href="/relative/path/">example</a>')
u'<a>example</a>'
```

I'd like to avoid this, since I want to retain all existing links (including relative path links).  But I'm not able to figure out how to do that -- I'm probably just missing something obvious!  Thanks for any suggestions.

**Comment by mauritsvanrees** [comment]
The same is happening now on PyPI.  The table of contents for packages is stripped of anchor links, which are also relative links.

This has been fixed in html5lib. From the changelog:

```
0.999999/1.0b7
Released on July 7, 2015

Fix #189: fix the sanitizer to allow relative URLs again (as it did prior to 0.9999/1.0b5).
```

**Comment by willkg** [comment]
This is fixed with the linkify rewrite.
